js/ui/iconGrid.js

```diff
@@ -157,10 +182,23 @@ const BaseIcon = new Lang.Class({
     _createIconTexture: function(size) {
         if (this.icon)
             this.icon.destroy();
+        this.extraIcons.forEach(function (i) {i.destroy()});
         this.iconSize = size;
         this.icon = this.createIcon(this.iconSize);
+        this.extraIcons = this.createExtraIcons(this.iconSize);
 
-        this._iconBin.child = this.icon;
+        this._layeredIcon.add_actor(this.icon);
+        if (this._shadowAbove) {
+            this._layeredIcon.set_child_below_sibling(this.icon, null);
+        }
```

Comment:
I believe in `eos-shell` there's some CSS used here to give this a more rounded lock. I'm not sure if we wanted to preserve those styles, so I've not included that here yet.